examples/generate_surrogate_test.py

```diff
@@ -0,0 +1,156 @@
+        if (t < 500):
+            i = 3
+        elif (t < 1000):
+            i = 2
+        elif (t < 1500):
+            i = 0.5
+        else:
+            i = 4.5
+        return batt.InputContainer({'i': i})
+    
+    def future_loading_2(t, x=None):
+        # Variable (piece-wise) future loading scheme 
+        if (t < 300):
+            i = 2
+        elif (t < 800):
+            i = 3.5
+        elif (t < 1300):
+            i = 4
+        elif (t < 1600):
+            i = 1.5
+        else:
+            i = 5
+        return batt.InputContainer({'i': i})
+    
+    load_functions = [future_loading_1, future_loading_2]
+
+    ## Step 3: generate surrogate model 
+    # Simulation options for training data and surrogate model generation
+    options_surrogate = {
+        'save_freq': 1, # For DMD, this value is the time step for which the surrogate model is generated
+        'dt': 0.1, # For DMD, this value is the time step of the training data
+        'data_len': 0.7, # Value between 0 and 1 that determines the fraction of data resulting from simulate_to_threshold that is used to train DMD surrogate model
+    }
+
+    # Set noise in Prognostics Model, default for surrogate model is also this value
+    batt.parameters['process_noise'] = 0
+
+    # Generate surrogate model  
+    DMD_approx = batt.generate_surrogate_dmd(load_functions,**options_surrogate)
+
+    ## Step 4: Implement surrogate model 
+    # Simulation options for implementation of surrogate model
+    options_sim = {
+        'save_freq': 1 # Frequency at which results are saved, or equivalently time step in results
+    }
+
+    # Define loading profile 
+    def future_loading(t, x=None):
+        if (t < 600):
+            i = 3
+        elif (t < 1000):
+            i = 2
+        elif (t < 1500):
+            i = 1.5
+        else:
+            i = 4
+        return batt.InputContainer({'i': i})
+
+    # Simulate to threshold using DMD approximation
+    simulated_results = DMD_approx.simulate_to_threshold(future_loading,**options_sim)
+
+    # Plot results
+    simulated_results.inputs.plot(ylabel = 'Current (amps)',title='Example 1 Input')
+    simulated_results.outputs.plot(ylabel = 'Predicted Outputs (temperature and voltage)',title='Example 1 Predicted Outputs')
+    simulated_results.event_states.plot(ylabel = 'Predicted State of Charge', title='Example 1 Predicted SOC')
+
+    # To visualize the accuracy of the approximation, run the high-fidelity model
+    options_hf = {
+        'dt': 0.1,
+        'save_freq': 1
+    }
+    high_fidelity_results = batt.simulate_to_threshold(future_loading,**options_hf)
+
+    # Save voltage results to compare
+    voltage_dmd = []
+    voltage_hf = []
+    for iter1 in range(len(simulated_results.times)):
+        voltage_dmd.append(simulated_results.outputs[iter1]['v'])
+    for iter2 in range(len(high_fidelity_results.times)):
+        voltage_hf.append(high_fidelity_results.outputs[iter2]['v'])
```

Comment:
Might be able to use a list comprehension here? Something like:
97. `voltage_dmd = [simulated_results.outputs[iter1]['v'] for iter1 in range(len(simulated_results.times))]`
98. `voltage_hf = [high_fidelity_results.outputs[iter2]['v'] for iter2 in range(len(high_fidelity_results.times))]`

Reply:
Ah this is exactly what I was trying to do - thank you @lawrence-hwang